Реализован функционал из домашнего задания спринта 12

DesidiamS opened the pull request.

Commit:
Реализован функционал из домашнего задания спринта 12

Commit:
Исправлен code-style

Commit:
Исправлен code-style

Commit:
Исправлен code-style

Commit:
Добавлено указание порта в application.properties

Commit:
Добавлено указание порта в application.properties

Commit:
Изменены способы создания таблиц и записей в них, чтобы не было дублей при повторном запуске приложения

Commit:
Убрано указание порта в application.properties

Commit:
Изменен способ создания БД чтобы не нарушать тесты postman

bismuthD commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,72 @@
+        primary key (FILM_ID, GENRE_ID)
+);
+
+create table if not exists USERS
```
Можно было бы ввести ограничение на уникальность логина и почты на уровне таблицы в БД

create unique index if not exists USER_EMAIL_UINDEX on USERS (email);
create unique index if not exists USER_LOGIN_UINDEX on USERS (login);

bismuthD commented on src/main/java/ru/yandex/practicum/filmorate/model/Genre.java:
```diff
@@ -0,0 +1,17 @@
+    Long id;
+    String name;
+
+    public Genre() {
```
зачем эти пустые конструкторы? можно добавить аннотацию @NoArgsConstructor, вместо них

bismuthD reviewed (commented):
Отличная работа, хорошо разобрался со взаимодействием между слоями сервис контроллер репозиторий, везде использованы верные модификаторы доступа, оставил пару комментариев на доработку

Commit:
- В моделях Genre и Mpa заменены пустые конструкторы на аннотацию @NoArgsConstructor
- В создании схемы БД были добавлены unique keys для полей email и login в таблице users
